src/doc/trpl/method-syntax.md

```diff
@@ -56,11 +58,115 @@ other parameter. Because we know it's a `Circle`, we can access the `radius`
 just like we would with any other struct. An import of π and some
 multiplications later, and we have our area.
 
+## Chaining method calls
+
+So, now we know how to call a method, such as `foo.bar()`. But what about our
+original example, `foo.bar().baz()`? This is called 'method chaining', and we
+can do it by returning `self`.
+
+```
+struct Circle {
+    x: f64,
+    y: f64,
+    radius: f64,
+}
+
+impl Circle {
+    fn area(&self) -> f64 {
+        std::f64::consts::PI * (self.radius * self.radius)
+    }
+
+    fn grow(&self) -> Circle {
+        Circle { x: self.x, y: self.y, radius: (self.radius * 10.0) }
+    }
+}
+
+fn main() {
+    let c = Circle { x: 0.0, y: 0.0, radius: 2.0 };
+    println!("{}", c.area());
+
+    let d = c.grow().area();
+    println!("{}", d);
+}
+```
+
+Check the return type:
+
+```
+# struct Circle;
+# impl Circle {
+fn grow(&self) -> Circle {
+# Circle } }
+```
+
+We just say we're returning a `Circle`. With this, we can grow a new circle
+that's twice as big as the old one. But what if we didn't want to make a new
+Circle, but change the old one? There's a pattern where this is commonly used:
+the builder pattern.
+
+## The Builder Pattern
+
+Let's say that we want our users to be able to create Circles, but we will
+allow them to only set the properties they care about. Otherwise, the `x`
+and `y` attributes will be `0.0`, and the `radius` will be `1.0`. Rust doesn't
+have method overloading, named arguments, or variable arguments. We employ
+the builder pattern instead. It looks like this:
+
+```
+struct Circle {
+    x: f64,
+    y: f64,
+    radius: f64,
+}
+
+impl Circle {
+    fn set_x(&mut self, x: f64) -> &mut Circle {
+	self.x = x;
+
+	self
+    }
+
+    fn set_y(&mut self, y: f64) -> &mut Circle {
+	self.y = y;
+
+	self
+    }
+
+    fn set_radius(&mut self, radius: f64) -> &mut Circle {
+	self.radius = radius;
+
```

Comment:
This is a personal / Ruby style thing that may be bleeding over. Do we have a convention here either? Basically, when doing a two or three liner, I'll make sure the final expression is on its own line so it's quite clear what's being returned. Maybe that's too much?